[cyli133] iP

cylithegreat opened the pull request:
### DukeCyli

> “Your mind is for having ideas, not holding them.” – David Allen [](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,

1. text-based
2. easy to learn
3. ~~FAST~~ SUPER FAST to use

All you need to do is:

- Download it from here [](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html)
- Double-click it.
- Add your tasks.
- Let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [X] Managing tasks.
- [ ] Managing deadlines (coming soon)
- [ ] Managing deadlines (coming soon) 

If you Java programmer, you can use it to practice Java too. Here's the main method:

`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}`

Commit:
Add Gradle support

cylithegreat committed:
Implemented level 1 for week 2

cylithegreat committed:
Implemented Level 2 for Week 2

cylithegreat committed:
Implemented Level 3 For Week 2

cylithegreat committed:
no message

cylithegreat committed:
Implemented Level 4 for Week 2.

cylithegreat committed:
Updated text-ui-test

cylithegreat committed:
Implemented Level 5 For Week 2.

cylithegreat committed:
Implemented Level 7 For Week 2

cylithegreat committed:
commits error in feature1 branch for testing

cylithegreat committed:
Tests the main branch merge

cylithegreat committed:
Merge branch 'master' into feature1

cylithegreat committed:
makes second commit to the feature branch

cylithegreat committed:
makes second commit to the main branch

cylithegreat committed:
Merge branch 'feature1'

cylithegreat committed:
add-coutnries commit 1

cylithegreat committed:
resume the test branches and commits

cylithegreat committed:
Implement the Save function, with adoptions of two functions provided by CHATGPT

cylithegreat committed:
Implement Level 8

cylithegreat committed:
Merge branch 'branch-level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java
#	src/main/java/Store.java
#	src/main/java/Tasks.java

cylithegreat committed:
Refactor classes

cylithegreat committed:
Separate the classes into packages

cylithegreat committed:
Implement A-Packages

cylithegreat committed:
Merge branch 'add-gradle-support' of https://github.com/cyli133/ip

cylithegreat committed:
Implement A-JUnit

cylithegreat committed:
Implement A-Jar

cylithegreat committed:
Implement A-JavaDoc

cylithegreat committed:
Implement A-JavaDoc.

cylithegreat committed:
Implemented A-CodingStandard.

cylithegreat committed:
Implements Find functionality for Level-9.

rmj1405 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,64 @@
+     * @throws DukeException
+     */
+    public static void createCommand (String input) throws DukeException {
+        switch (input){
```
Nice to see that you have avoid deep nesting in your code 👍

rmj1405 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,64 @@
+    public static void createCommand (String input) throws DukeException {
+        switch (input){
+            default : {
+                if (input.matches("mark+ [0-9]+")) {
```
Perhaps these if-else statements can be simplified or abstracted out to simplify the code?

rmj1405 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,64 @@
+                    queue.add(new Delete(input));
+                } else if (input.matches("find\\s.*$")){
+                    queue.add(new Find(input));
+                } else if (input.matches("^deadline\\s.*$") || input.matches("^event\\s.*$") || input.matches("^todo\\s.*$")) {
```
Cool regex! What does "^deadline\\s.*$" mean?

rmj1405 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,51 @@
+     * @param task The ArrayList of Tasks to store in the local hard disk.
+     */
+    //Credits: Adapted from CHATGPT
+    public void saveTasks(ArrayList<Tasks> task) {
```
Nice distinguishing between single-valued and multi-valued objects/naming 👍

rmj1405 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,34 @@
+ */
+public class Duke {
+
+    public static boolean offBot = false;
```
Perhaps you would like to rename this to sound more like a boolean i.e. isOffBot?

rmj1405 reviewed (commented):
Overall neat code! Perhaps you might want to consider abstracting out the if-else statements to improve readability 😀

cylithegreat committed:
Implements personal modificaitons.

cylithegreat committed:
Implements search by date.

Johnchiahk commented on src/main/java/duke/command/Delete.java:
```diff
@@ -0,0 +1,27 @@
+/**
+ * Implements the delete command.
+ */
+public class Delete extends Commands{
```
Perhaps a space after Commands would be better?

Johnchiahk commented on src/main/java/duke/command/TaskCreationCommands.java:
```diff
@@ -0,0 +1,41 @@
+        this.setCommandStorage(str);
+    }
+
+    /**
+     * Executes the task creation commands stored in the command queue, by parsing the user input.
+     * @param list the list of tasks to execute the commands on.
+     */
```
Nice! I like how every JavaDocs of each method is very well defined 👍

cylithegreat committed:
Add missing branches required by the admin tasks. Implement enum. Implement partial search by date. Refactor some classes.

cylithegreat committed:
Implement level-10, refactor code for better structure and complete some bug fixes

cylithegreat committed:
Implement A-CodeQuality

cylithegreat committed:
Update .gitignore

cylithegreat committed:
Implement A-CodingStandard and Enum

cylithegreat committed:
Fix minor bugs

cylithegreat committed:
Implement A-CodeQuality

cylithegreat committed:
Implement Asserts

cylithegreat committed:
Implement Assert 2

cylithegreat committed:
Merge pull request #2 from cyli133/branch-A-CodeQuality

Implement A-CodeQuality

cylithegreat committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/duke/command/Mark.java

cylithegreat committed:
Merge pull request #3 from cyli133/branch-A-Assertions

Implement Assert 2

cylithegreat committed:
Implement varargs

cylithegreat committed:
New class Update: Implement update dates functionality

The bot does not allow users to update event time. This posts potential
inconveniences to the users. Firstly, if users keyed in the wrong date,
they have to delete the task and enter a new one. Secondly, if the user
decides to postpone the task, they have to recreate the task as well.

Let's implement a new function to allow users to update the dates of
an existing task.

Refer to this requirements for the specific requirements:
https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/
projectDuke/index.html#level-4-todos-events-deadlines

cylithegreat committed:
Update Functionality: Enables wrong data format detection

The bot does not detect the wrong user format for the update command,
it crashes when user inputs random strings after the update keyword.
This disrupts the user experience, and leave the user not knowing what
is happening and what is supposedly the correct format.

Let's implement a user input validation function that tells the users
when they key in the wrong data format for the update command.

Refer to this requirements for the specific requirements:
https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/
projectDuke/index.html#level-4-todos-events-deadlines

cylithegreat committed:
Bye Class: Fix bug of program not exiting

The bot does not exit after showing the "bye" message. This defeats the
purpose of having a bye command if users can still key in new commands
after the Bye command.

Let's allow the program to fully exit after receiving the Bye command.

cylithegreat committed:
Improve the Javadoc and the code appearance

cylithegreat committed:
Implement minor style fixes

cylithegreat committed:
Implement minor bug fix for todo update

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Update README.md

cylithegreat committed:
Write user guide

cylithegreat committed:
Fix minor bugs

cylithegreat committed:
Update Readme

cylithegreat committed:
Improve READ.me

cylithegreat committed:
Add demo picture

cylithegreat committed:
Update build.gradle
